tests/read-only-store.sh

```diff
@@ -0,0 +1,29 @@
+source common.sh
+
+clearStore
+
```

Comment:
```suggestion

happy () {
    # We can do a read-only query just fine with a read-only store
    nix path-info $dummyPath
    
    # We can "write" an already-present store-path a read-only store, because no IO is actually required
    nix-store --store local?read-only=true --add dummy
}
```